src/Microsoft.ML.Data/Transforms/HashTransform.cs

```diff
@@ -597,6 +687,58 @@ private static uint HashCore(uint seed, ref DvText value, int i, uint mask)
             return (value.Trim().Hash(Hashing.MurmurRound(seed, (uint)i)) & mask) + 1;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static uint HashCore(uint seed, ref float value, uint mask)
+        {
+            Contracts.Assert(Utils.IsPowerOfTwo(mask + 1));
+            if (value.IsNA())
+                return 0;
+            // (value == 0 ? 0 : value) takes care of negative 0, its equal to positive 0 according to the IEEE 754 standard
```

Comment:
Do we want to treat `+0` and `-0` as equal for the purposes of hashing?

Reply:
2 NaNs - do u mean minus infinity and plus infinity? I would say they are pretty far apart :) unless we are in  Lobachevskian geometry, so they should be end up with different hash

Reply:
No, I am referring to `double.NaN`, for which `double.NaN` is one representation, but for which there are some 16.7 million other representations (same applies to `float.NaN`, but with fewer total representations).

Two `NaN` never compare as equal, and if we were to normalize any values, I would expect those to be the only ones we do normalize.

I would not expect normalization of `-0`, as that is an important and distinguishing number for IEEE math.

Reply:
Talked to Tom, every NaN will be handled as NA, so hashed to 0